source/common/http/filter/transformation_filter.cc

```diff
@@ -153,32 +165,49 @@ void FunctionalTransformationFilter::checkRequestActive() {
 
 void TransformationFilterBase::checkResponseActive() {
   response_transformation_ = getTransformFromRoute(
-      encoder_callbacks_->route(),
-      Config::MetadataTransformationKeys::get().RESPONSE_TRANSFORMATION);
+      encoder_callbacks_->route(), TransformationFilterBase::Direction::Response);
 }
 
 const envoy::api::v2::filter::http::Transformation *
 TransformationFilterBase::getTransformFromRoute(
-    const Router::RouteConstSharedPtr &route, const std::string &key) {
+    const Router::RouteConstSharedPtr &route, TransformationFilterBase::Direction direction) {
 
   if (!route) {
     return nullptr;
   }
 
+  const auto *config = SoloFilterUtility::resolvePerFilterConfig<
+      RouteTransformationFilterConfig>(
+      Config::TransformationFilterNames::get().TRANSFORMATION, route);
+
+  if (config != nullptr) {
+    switch (direction) {
+    case TransformationFilterBase::Direction::Request:
+      return &config->getRequestTranformation();
+    case TransformationFilterBase::Direction::Response:
+      return &config->getResponseTranformation();
+    }
+  }
+
+  if (config_ == nullptr) {
+    // this can happen in per route config mode.
+    return nullptr;
+  }
```

Comment:
1. I find the fact that a comparison like `(config != nullptr)` is followed by `(config_ == nullptr)`, which refers to a different identifier that looks very similar, a bit confusing. Is this intentional?
2. IIUC the per route config mode is immutable and set during construction. If that it the case, consider implementing a different sub-class for each of the two modes, so that specific member functions do not need to support both modes and repeat the mode check upon every invocation.

Reply:
1. one is a member (config form filter config) the other is local var for config from route.
2. the reason i prefer to avoid that is because we plan to remove static filter config all together once this is implemented in gloo..